extensions/amp-viafoura/0.1/amp-viafoura.js

```diff
@@ -38,40 +40,64 @@ class AmpViafoura extends AMP.BaseElement {
     return isLayoutSizeDefined(layout);
   }
 
+  getDataAttrs() {
+    return Array.prototype.slice.call(this.element.attributes)
+      .map(attr => attr.nodeName)
+      .filter(attrName => attrName.match('data-.*'))
+      .map(name => [name, this.element.getAttribute(name)])
+  }
+
+  getPageAttrs() {
+    var attributes = locations();
+    attributes.push(['url', location.href])
+    return loc;
+  }
+
+  getParams() {
+    return this.getDataAttrs()
+      .concat(this.getPageAttrs())
+      .map(pair => `${pair[0]}=${encodeURIComponent(pair[1])}`)
+      .reduce((prev, cur) => prev ? `${prev}&${cur}` : cur);
+  }
+
   /** @override */
   layoutCallback() {
-    const getAttr = attr => encodeURIComponent(this.element.getAttribute(attr));
-    const widget = getAttr("data-widget");
-    const limit = getAttr("data-limit");
-    const sort = getAttr("data-sort");
-    const path = getAttr("data-path");
-    const title = getAttr("data-title");
-    const unique_id = getAttr("data-unique-id");
 
     const width = this.element.getAttribute('width');
     const height = this.element.getAttribute('height');
 
     const iframe = document.createElement('iframe');
     iframe.setAttribute('frameborder', '0');
-    iframe.src = `${FRAME_SRC}?limit=${limit}&sort=${sort}&path=${path}&title=`+
-        `${title}&unique_id=${unique_id}&widget=${widget}`;
+    iframe.src = `${FRAME_SRC}?${this.getParams()}`;
 
     this.applyFillContent(iframe);
 
     iframe.width = width;
     iframe.height = height;
+    iframe.setAttribute('scrolling', 'no');
+
     this.element.appendChild(iframe);
 
     /** @private {?Element} */
     this.iframe_ = iframe;
 
+    var currentHeight = 0;
+    listen(iframe, 'embed-size', function(data) {
+
+      if (currentHeight !== data.height) {
+        // Extra padding for the comment share dropdown
+        currentHeight = data.height;
+        this.element.style.height = data.height + 100 + 'px';
```

Comment:
No, this is due to issues in how we are handling the sizing on the Front end, but hopefully can be eventually removed. Right now, we trigger the resize events before animations are complete, so we need some extra padding. There also seem to be some issues even when using the "proper" height.